### PR TITLE
docs: considerations on block storage solutions

### PR DESCRIPTION
**Pull request by weiyentan**
This adds to the documentation for storage to document the feature of data-locality highlighting the scenarios where instances of the data could be on the same node. This can add a risk because if the node is destroyed (as per treating nodes like cattle) the cluster can be corrupted. The documentation highlights this and suggests certain possible solutions that can be investigated.

### CODE REVIEW

File: docs/src/storage.md [hunk 1]
```diff
@@ -467,16 +467,43 @@ To use a pre-provisioned volume in CloudNativePG:
...
+approach works well for workloads that lack built-in resiliency.
+
+However, CloudNativePG integrates this resiliency directly into the Postgres
+`Cluster` through the number of instances and the persistent volumes attached
+to them, as explained in ["Synchronizing the state"](architecture.md#synchronizing-the-state).
+
+As a result, defining additional replicas at the storage level can lead to
+write amplification, unnecessarily increasing disk I/O and space usage.
+
+In such cases, consider reducing the number of replicas at the block storage
```
Comment (jsilvela) [inline docs/src/storage.md:483]: I don't love "in such cases" here. Do we mean "cases where the application handles resiliency above the block level"? Or do we mean, "when using CloudNativePG"?
```suggestion
For CloudNativePG usage, consider reducing the number of replicas at the block storage
```

**Review by jsilvela — COMMENTED** [review]
This is a very good document @weiyentan , thanks.
I left an inline suggestion.
And, for completeness, perhaps we could discuss the case where a single instance is used? Obviously not recommended to use a single instance, but if there were a need? Should block-level redundacy be used?

**Comment by weiyentan** [comment]
Happy to talk about this. @jsilvela

File: docs/src/storage.md (hunk 1 above)
Comment (gbartolini) [reply docs/src/storage.md:483]: I suggest you directly make the change @jsilvela so that we can have this doc in the next release. Thanks.

**Review by gbartolini — APPROVED** [review]

**Review by jsilvela — APPROVED** [review]

**Comment by jsilvela** [comment]
Made the change and approved.
@gbartolini I don't have auth to merge this, can you?
Oh, on @weiyentan 's commits, the signoff is missing. See the DCO failure report:
> Commit sha: [a62fbe9], Author: Wei-Yen Tan, Committer: Gabriele Bartolini; The sign-off is missing.
Commit sha: [45d8fb4], Author: Wei-Yen Tan, Committer: Gabriele Bartolini; The sign-off is missing.
Commit sha: [1b86b2c], Author: Wei-Yen Tan, Committer: Gabriele Bartolini; The sign-off is missing.

**Pull request merged**